### DISCUSSION TITLE
dynamic configuration wdio.conf.js

### DISCUSSION THREAD

**Discussion by DELEPaps** [webdriverio/webdriverio#8901] [Q&A] [answered]
I want to use the WDIO testrunner API, but can I pass additional arguments and access them inside wdio.conf.js

https://webdriver.io/docs/setuptypes#package-api-2

**Comment by christian-bromann** [comment, marked as answer]
Yes, see the example:

```js
const wdio = new Launcher(
    '/path/to/my/wdio.conf.js',
    { spec: '/path/to/a/single/spec.e2e.js' }
)
```

The second argument allows to pass in arguments. Note: this doesn't work for hooks.

**Reply by DELEPaps** [reply]
yes i know, but i mean extra argument that i can call inside wdio.conf.js file

e.g. I want dynamic values for capabilities chrome profiles.
example:
```
PROFILE=1 wdio run /path/to/my/wdio.conf.js --spec /path/to/a/single/spec.e2e.js
```
I can access the PROFILE env variable in the wdio.conf.js file

can i get that with Launcher?

**Reply by christian-bromann** [reply]
yes

**Reply by DELEPaps** [reply]
can you provide an example using Launcher? and access it in wdio.conf.js . file

**Reply by christian-bromann** [reply]
I don't have an example but you can always use things like `process.argv` to access CLI arguments.

**Reply by DELEPaps** [reply]
i know that, my question is this can be implemented with class Launcher?
```
where can i put my own argument? instead of 'PROFILE=1`
const wdio = new Launcher(
    '/path/to/my/wdio.conf.js',
    { spec: '/path/to/a/single/spec.e2e.js' }
)
```

**Reply by christian-bromann** [reply]
And again, yes!

**Reply by DELEPaps** [reply]
Sorry maybe my explanation makes you don't understand what I mean.
i build electron application and call Launcher class from ipcMain.

you suggested this before (https://github.com/webdriverio/webdriverio/issues/1914#issuecomment-285668885), but it's too much trouble for me, at least for now.

After digesting your suggestion above, I was wondering if in nodejs can create a temporary env?
and at least this solved my problem.
```
// ipcMain
process.env = {
    // ...process.env,
    PROFILE: 5,
};

const wdio = new Launcher(
    '/path/to/my/wdio.conf.js',
    { spec: '/path/to/a/single/spec.e2e.js' }
)
```
thanks for helping.
